src/libclient/net/sessionlistingmodel.cpp

```diff
@@ -196,35 +229,60 @@ QVariant SessionListingModel::data(const QModelIndex &index, int role) const
 
 QVariant SessionListingModel::headerData(int section, Qt::Orientation orientation, int role) const
 {
-	if(role != Qt::DisplayRole || orientation != Qt::Horizontal)
+	if(orientation != Qt::Horizontal)
 		return QVariant();
 
-	switch(section) {
-	case ColumnVersion: return QVariant();
-	case ColumnStatus: return QVariant();
-	case ColumnTitle: return tr("Title");
-	case ColumnServer: return tr("Server");
-	case ColumnUsers: return tr("Users");
-	case ColumnOwner: return tr("Owner");
-	case ColumnUptime: return tr("Age");
+	if (role == Qt::AccessibleTextRole && section == Version) {
+		return tr("Compatibility");
+	} else if (role != Qt::DisplayRole) {
+		return QVariant();
 	}
-	static_assert (ColumnCount == 7, "Update headerData()");
 
+	switch(Column(section)) {
+	case Title: return tr("Title");
+	case Server: return tr("Server");
+	case UserCount: return tr("Users");
+	case Owner: return tr("Owner");
+	case Uptime: return tr("Age");
+	case Version:
+	case ColumnCount: {}
+	}
 	return QVariant();
 }
 
 Qt::ItemFlags SessionListingModel::flags(const QModelIndex &index) const
 {
-	if(index.internalId()>0) {
-		const Listing &l = m_listings.at(int(index.internalId() - 1));
-		if(!l.message.isEmpty())
-			return Qt::NoItemFlags;
-
-		const Session &s = l.sessions.at(index.row());
-		if(!s.protocol.isCompatible())
-			return Qt::NoItemFlags;
+	if (isRootItem(index)) {
+		return Qt::ItemIsEnabled;
+	}
+
+	const auto &listing = m_listings.at(listingIndex(index));
+	if(listing.offline())
+		return Qt::ItemIsEnabled | Qt::ItemNeverHasChildren;
+
+	const auto &session = listing.sessions.at(index.row());
+	if(!session.protocol.isCurrent())
```

Comment:
```suggestion
	if(!session.protocol.isCompatible())
```

Otherwise you can't join sessions in compatibility mode.

Reply:
This is fixed now.